apps/devhub/views.py

```diff
@@ -1196,6 +1197,19 @@ def version_add(request, addon_id, addon):
                 form.cleaned_data['nomination_type']):
             addon.update(status=form.cleaned_data['nomination_type'])
         url = reverse('devhub.versions.edit', args=[addon.slug, str(v.id)])
+
+        if not addon.is_listed:  # Not listed? Do we need to sign?
+            # We need to get to the file to check its validation.
+            # We're submitting a new version here, so it has only one file.
+            file_ = v.files.get()
```

Comment:
I've done it (the cleaning up) from time to time, but I'd rather do it each time, it's a good idea ;) I'll change that.

And as you've commented below, in this case we're adding a new version, with only one file (at this point). Later, once the version has been created, it's possible to add new files to it (dealt with in `version_add_file`).